### PR TITLE
Combo fix

### PR DESCRIPTION
**Pull request by somehibs**
Major changes are
 - removed .idea folder - this is generated by Gradle when opening the project for the first time
 - added a CombinationActivity which uses the tripbot api rather than the combos.json object
 - a new combination category (caution) will need adding to separate translation documents
 - changed some style parents to use appcompat styles - themes unaffected
 - changed activity parents to use appcompat activity - brings in bugfixes for older platforms

### CODE REVIEW

File: TripMobile/app/src/main/java/me/tripsit/mobile/views/BackgroundAnimation.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        this.viewsToAnimate = viewsToAnimate;
+    }
+
+    public void init(Context ctx, AttributeSet attrs) {
+    }
+
+    public synchronized void setBackgroundColor(int i) {
+        setColorImmediate(i);
+
+        // finish the rest off later
```
Comment (EddieCurtis) [inline TripMobile/app/src/main/java/me/tripsit/mobile/views/BackgroundAnimation.java]: Do we need this code?

File: TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+        BackgroundAnimation animation = new BackgroundAnimation(viewsToAnimate);
+        animation.setBackgroundColor(color);
+
+        header.setVisibility(View.VISIBLE);
+        content.setVisibility(View.VISIBLE);
+    }
+
+    Button search = null;
+    private void testCombinationActivity() {
```
Comment (EddieCurtis) [inline TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java]: Did you want to leave this here or can we remove it?

File: TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                    drug = getString(R.string.failed_empty_drug);
+                }
+                error = String.format(getString(R.string.failed_search_combo), drug);
+                break;
+            case NETWORK_ERROR:
+                error = getString(R.string.failed_download_combinations);
+                break;
+        }
+
+        /*AlertDialog dialog = new AlertDialog.Builder(this)
```
Comment (EddieCurtis) [inline TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java]: Can this be removed?

File: TripMobile/app/src/main/java/me/tripsit/mobile/views/BackgroundAnimation.java (hunk 1 above)
Comment (somehibs) [inline TripMobile/app/src/main/java/me/tripsit/mobile/views/BackgroundAnimation.java]: Not needed but rather have it there for someone to improve than not

File: TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java (hunk 2 above)
Comment (somehibs) [inline TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java]: I'd prefer to leave it, I like a convenient way of testing a view like this.

File: TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java (hunk 3 above)
Comment (somehibs) [inline TripMobile/app/src/main/java/me/tripsit/mobile/combination/CombinationActivity.java]: Done

**Pull request merged**